internal/gitinterface/hash.go

```diff
@@ -23,17 +23,17 @@ var (
 // Hash represents a Git object hash. It is a lightweight wrapper around the
 // standard hex encoded representation of a SHA-1 or SHA-256 hash used by Git.
 
-type Hash []byte
```

Comment:
Why are we setting the size of the hash to sha1.Size? Wouldn't this go against hash agility, as specified in the docstring?

Reply:
Good catch! This will truncate the hash down to the sha1 size. The tests don't catch this, since the assertions are only comparing the truncated values.

We should revert this, plus this to the test to make sure we can roundtrip the original hash back to the string we gave it.

```go
assert.Equal(t, test.hash, hash.String())
```

Reply:
good catch, this was to simplify Hash comparison elsewhere as well as to be able to use the hash obj as a map key. For now, i'm using the string representation of a hash for the map key and I've added a .Equal() on hash for the comparisons.

Reply:
@wlynch added the roundtrip check too.